ModuleSettings.cs

```diff
@@ -83,26 +89,26 @@ public class ModuleSettings {
 
         public SettingCollection MapSettings { get; private set; }
 
-        public SettingEntry<bool> MapPathablesEnabled                   { get; private set; }
-        public SettingEntry<bool> MapShowMarkersOnFullscreen            { get; private set; }
-        public SettingEntry<bool> MapShowTrailsOnFullscreen             { get; private set; }
-        public SettingEntry<bool> MapShowMarkersOnCompass               { get; private set; }
-        public SettingEntry<bool> MapShowTrailsOnCompass                { get; private set; }    
-        public SettingEntry<bool> MapShowAboveBelowIndicators           { get; private set; }
-        public SettingEntry<bool> MapFadeVerticallyDistantTrailSegments { get; private set; }
+        public SettingEntry<bool>            MapPathablesEnabled                   { get; private set; }
+        public SettingEntry<VisibilityLevel> MapShowMarkersOnFullscreen            { get; private set; }
+        public SettingEntry<VisibilityLevel> MapShowTrailsOnFullscreen             { get; private set; }
+        public SettingEntry<VisibilityLevel> MapShowMarkersOnCompass               { get; private set; }
+        public SettingEntry<VisibilityLevel> MapShowTrailsOnCompass                { get; private set; }
```

Comment:
Let's rename these to better represent their new full purpose.  In any case, we need new names because the settings are stored with these names between versions and any users attempting to load their settings that were previously bools are likely to experience their settings failing to load because of the type change.  Best to rename and avoid the conflict anyways.